### ISSUE TITLE
BinaryNetConv2D 

### ISSUE THREAD

**Issue by mamba36** [pieterluitjens/BinaryNet_and_XNORNet#5]
Why is the value of feature map after the BinaryNetConv2D layer not integer? The result of the convolution between binarized weights and binarized input should be integer. But when I run your code, I find that the feature maps is all decimals.

**Comment by yaysummeriscoming** [comment]
For backprop to work, training needs to be done in float precision

**Comment by mamba36** [comment]
> 
> 
> For backprop to work, training needs to be done in float precision

Thank you very much for your answer.But my problem is in the feedforward phase。The features and weights are both binary, the convolution result should be integer, but I'll run your code to see that the intermediate result is floating point. The image shows the binary_net_conv2d_2 layer's features. Its supposed to be an integer.
![图片](https://user-images.githubusercontent.com/37042167/87857093-96a53000-c956-11ea-92c9-2e48331db6af.png)

**Comment by yaysummeriscoming** [comment]
Yes, but the forward pass also needs to be done in float precision for backprop to work. For binary inference, you'll need to use another software package.

**Comment by mamba36** [comment]
Can you tell me which package I should use？




&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "yaysummeriscoming/BinaryNet_and_XNORNet"                                                                                    <notifications@github.com&gt;;
发送时间:&nbsp;2020年7月20日(星期一) 晚上9:37
收件人:&nbsp;"yaysummeriscoming/BinaryNet_and_XNORNet"<BinaryNet_and_XNORNet@noreply.github.com&gt;;
抄送:&nbsp;"mamba36"<1021749442@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [yaysummeriscoming/BinaryNet_and_XNORNet] BinaryNetConv2D  (#5)





 
Yes, but the forward pass also needs to be done in float precision for backprop to work. For binary inference, you'll need to use another software package.

**Comment by yaysummeriscoming** [comment]
Unfortunately I haven't kept up to date with the latest developments in binary networks - it's been a few years since I made this. That being said, I've heard that plumerai & mxnet have packages available
